### ISSUE TITLE
Fix false positives for multiline complex selectors in indentation

### ISSUE THREAD

**Issue by artola** [stylelint/stylelint#3286]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug related to long selectors when breaking in multiline.

> Which rule, if any, is this issue related to?

`indentation` and `selector-combinator-space-before`

> What CSS is needed to reproduce this issue?

```css
.foo-long-long-long-long-name
  > .bar-long-long-long-long-name
  > .baz-long-long-long-long-name {
  color: red;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "indentation":  2,
    "selector-combinator-space-before": "always"
  }
}
```

> Which version of stylelint are you using?

`9.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint --config myconfig *.css`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

It happens with CSS as well with SCSS.

> What did you expect to happen?

Indentation rule should not run inside the selector. Selector combionator should consider "always" as 1 or more spaces.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
test.css
 2:3   ✖  Expected indentation of 0 spaces   indentation
 2:3   ✖  Expected single space before ">"   selector-combinator-space-before
 3:3   ✖  Expected indentation of 0 spaces   indentation
 3:3   ✖  Expected single space before ">"   selector-combinator-space-before
```

> Note:

The code is formatted by Prettier, and I agree that breaking long selectors as Prettier does is more readable than removing the leading indentation as suggested by this rule.

**Comment by jeddy3** [comment]
Closing for now. We can always revisit if there are more requests.
